### ISSUE TITLE
integration test: write /sys/fs/cgroup/xxx/cgroup.subtree_control: no such file or directory

### ISSUE THREAD

**Issue by sctb512** [containerd/nydus-snapshotter#490]
Currently, the e2e test uses `ubuntu-latest` (i.e. ubuntu-22.04) image, which enables cgroup v2 only. This leads the behavior of `nydus-snapshotter` or `containerd` running in a container is not as expected.
Although the e2e test was successful, I reviewed the integration log and found the following logs:
```
time="2023-05-22T07:52:12.611032385Z" level=error msg="failed to enable controllers ([cpuset cpu io memory hugetlb pids
 rdma misc])" error="failed to write subtree controllers [cpuset cpu io memory hugetlb pids rdma misc] to 
\"/sys/fs/cgroup/default/cgroup.subtree_control\": write /sys/fs/cgroup/default/cgroup.subtree_control: 
no such file or directory"
time="2023-05-22T07:52:12.611160686Z" level=warning msg="error from *cgroupsv2.Manager.EventChan" 
error="failed to add inotify watch for 
\"/sys/fs/cgroup/default/d5f94ffbcc62095d9e1232ebabe9a31d2efbc5e84e90a2855f25661231ec7a68/memory.events\": 
no such file or directory"
```

The reason for this issue is cgroup v2 enables controllers for children via `cgroup.subtree_controllers`. If we add "+memory" to this file, we should make sure it already exists in the root group (`/sys/fs/cgroup/cgroup.subtree_controllers`). In container, the value of `/sys/fs/cgroup/cgroup.subtree_controllers` is null even if it is as expected in the host .

```shell
❯ nerdctl run --rm -it golang
root@0773a000af6b:/go# cd /sys/fs/cgroup
root@0773a000af6b:/sys/fs/cgroup# cat cgroup.subtree_control
root@0773a000af6b:/sys/fs/cgroup# exit
exit
❯ pwd
/sys/fs/cgroup
❯ cat cgroup.subtree_control
cpuset cpu io memory pids
```

Currently, I am working for [`limit nydus daemon resource limitation via cgroup`](https://github.com/containerd/nydus-snapshotter/pull/415) and facing the same problem.

The potential solution to this problem:
1. Write controllers to `/sys/fs/cgroup/cgroup.subtree_controllers`. I tried this, but it was not successful because the existed processes were added to the root group. 
2. Try to move processes in the root cgroup to children. ~~It doesn't work for me as well.~~ It works, but the processes moved to child group can not be moved back after writing `/sys/fs/cgroup/cgroup.subtree_controllers`.
```shell
echo: write error: Invalid argument
echo: write error: Invalid argument
echo: write error: Invalid argument
echo: write error: Invalid argument
echo: write error: Invalid argument
echo: write error: Invalid argument
echo: write error: Invalid argument
echo: write error: Invalid argument
```

Refer to https://github.com/moby/moby/issues/43093 and https://unix.stackexchange.com/questions/680167/ebusy-when-trying-to-add-process-to-cgroup-v2

So, I believe we need to modify the e2e test configuration to resolve this issue.

**Comment by sctb512** [comment]
Fixed by #415 

To resolve this issue, two steps are sufficient.
1. set `cgroup-parent` for container.
```shell
docker run --cap-add SYS_ADMIN --security-opt seccomp=unconfined --cgroup-parent=system.slice 
--cgroupns private --name nydus-snapshotter_e2e --rm --privileged golang
```
2. enable nesting for cgroup v2, i.e. move processes in root group to children `init` and write controllers to `/sys/fs/cgroup/cgroup.subtree_control`.
```shell
if [ -f /sys/fs/cgroup/cgroup.controllers ]; then
    mkdir -p /sys/fs/cgroup/init
    xargs -rn1 < /sys/fs/cgroup/cgroup.procs > /sys/fs/cgroup/init/cgroup.procs || :
    sed -e 's/ / +/g' -e 's/^/-/' < /sys/fs/cgroup/cgroup.controllers \
        > /sys/fs/cgroup/cgroup.subtree_control
fi
```

**Comment by sctb512** [comment]
#415
